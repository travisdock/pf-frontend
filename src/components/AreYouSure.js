import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.onOpen}
            >
            <div className="table-popup">
                <p>{this.props.message}</p>
                <button className="delete" onClick={this.props.handleDelete} >Delete</button>
                <button className="close" onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;