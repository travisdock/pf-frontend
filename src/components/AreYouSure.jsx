import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.askIfSure}
            >
            <div className="table-popup">
                <p>Are You Sure?</p>
                <button onClick={this.props.handleDelete} >Delete</button>
                <button onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;