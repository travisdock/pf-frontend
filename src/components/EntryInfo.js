import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button className="delete" onClick={this.props.askIfSure}>Delete Entry</button>
                <button className="update" onClick={this.props.openUpdateModal}>Update Entry</button>
                <button className="copy" onClick={() => this.props.copyEntry(this.props)}>Copy Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;