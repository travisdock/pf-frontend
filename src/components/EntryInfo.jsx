import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button onClick={this.props.askIfSure}>Delete Entry</button>
                <button onClick={this.props.openUpdateModal}>Update Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;