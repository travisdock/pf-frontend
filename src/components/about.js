import React from 'react';

const About = () => (
  <div className="about">
    <h1>Sum.</h1>
    <p>Some guy once said something like "What gets measured, gets managed". For people having trouble managing their personal finances, simply measuring their income and expenses is the first step. Often just the act of knowing where your money is going brings about the motivation to make changes. Sum. is a very simple app that gives the user a lot of freedom in simply measuring. Create any number of custom categories for income and expenses. Then watch where the money goes. Once you actually see how your expenses add up it will almost be harder not to make any changes.</p>
  </div>
)

export default About;
