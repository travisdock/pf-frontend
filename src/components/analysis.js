import React from 'react';
import { connect } from 'react-redux';

import Chart from './chart';

class Analysis extends React.Component {
  state = {
    userEntries: []
  }

  componentDidMount() {
    const url = `http://localhost:3001/api/v1/month_category/${this.props.current_user.user_id}`
    fetch(url)
      .then(resp => resp.json())
      .then(resp => this.setState({userData: resp}))
  }
  render(){

    return(
      <div>{this.state.userData ?
        <div><Chart /></div>
        :
        <div>Loading...</div>
      }</div>
    )
  }
}

const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Analysis);
