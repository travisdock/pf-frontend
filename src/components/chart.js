import React from 'react';
import {bb} from "billboard.js";
import { connect } from 'react-redux';

class Chart extends React.Component {

  state = {
    charts: {},
    currentChart: 0,
    stats: {},
    extraMonthInfo: {},
    load: false
  }

  componentDidMount() {
    const id = this.props.current_user.user_id

    fetch(`${process.env.REACT_APP_API}/api/v1/charts/${id}`)
      .then(resp => resp.json())
      .then(resp => {if (resp.error) {
        this.setState({error: resp.error})
      } else {
        const titles = Object.keys(resp.charts)
        const lastMonth = titles[titles.length - 1]
        this.setState({
          charts: resp.charts,
          stats: resp.stats,
          load: true,
          currentChart: resp.charts[lastMonth], currentTitle: lastMonth
        }, this.renderPieChart)
      }}
        )
  }

  selectChangePieChart = (e) => {
    const title = e.target.value
    this.setState( () => {
      return {
        currentChart: this.state.charts[title],
        currentTitle: title
      }
    }, this.renderPieChart)
  }

  renderPieChart() {
    const { currentChart, currentTitle } = this.state
    // const { currentTitle } = Object.keys(currentChart)[0]
    // debugger;
    bb.generate({
      data: {
        columns: currentChart,
        type: "pie",
        // onclick: function(d, element) { debugger }
      },
      title: {
        text: currentTitle,
        position: "top-center",
        padding: {
            top: 10,
           right: 10,
            bottom: 10,
            left: 10
        },
      },
      legend: {
        show: true,
        // contents: {
        //     bindto: "#legend",   // <ul id='legend'></ul>
        //     template: "<li style='background-color:#1f77b4'>data1</li>"
        //     or using function
        //     template: function(id, color, data) {
        //          // if you want omit some legend, return falsy value
        //          if (title !== "data1") {
        //               return "<li style='background-color:"+ color +">"+ title +"</li>";
        //          }
        //     }
        // },
        position: "bottom",  // bottom, right, inset
        equally: false,
        padding: 10,
        item: {
            // onclick: function(id) { debugger },
            // onover: function(id) { debugger },
            // onout: function(id) { debugger },
    
            // set tile's size
            tile: {
                width: 20,
                height: 15
            }
        }
      },
      bindto: "#chart"
    });
  }

  render() {
    return (
      <div className="content">
        {this.state.load ?
            <div className="chart-content">
              <select
                name="chart"
                value={this.state.currentTitle} //current chart month
                onChange={this.selectChangePieChart}
                >
                {Object.keys(this.state.charts).map(title => <option value={title} key={title} > {title} </option>) }
              </select>
              <div id="chart" />
            </div> :
            this.state.error ? <div>{this.state.error}</div> : <div>Loading...</div>
        }
      </div>
    )
  }
}
const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Chart)



// This is the profit loss chart. I got rid of it because it was ugly but I may want to remake it one day so here is the code.
// renderPLChart() {
//   const { currentChart } = this.state
//   bb.generate({
//     data: {
//       x: "x",
//       columns: currentChart,
//     type: "area-spline"
//     },
//     axis: {
//       x: {
//         type: "timeseries",
//         tick: {
//           format: "%b"
//         }
//       }
//     },
//     bindto: "#chart"
//   });
// }