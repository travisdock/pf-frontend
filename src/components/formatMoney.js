//https://gist.github.com/hengkiardo/3760884
//I took this function from ^this gist but changed the j declaration
//we will see if it breaks
export const formatMoney = (number, places, symbol, thousand, decimal) => {
    number = number || 0;
    places = !isNaN(places = Math.abs(places)) ? places : 2;
    symbol = symbol !== undefined ? symbol : "$";
    thousand = thousand || ",";
    decimal = decimal || ".";
    var negative = number < 0 ? "-" : "",
        i = parseInt(number = Math.abs(+number || 0).toFixed(places), 10) + "",
        j = i.length > 3 ? i.length % 3 : 0;
    return symbol + negative + (j ? i.substr(0, j) + thousand : "") + i.substr(j).replace(/(\d{3})(?=\d)/g, "$1" + thousand) + (places ? decimal + Math.abs(number - i).toFixed(places).slice(2) : "");
  }