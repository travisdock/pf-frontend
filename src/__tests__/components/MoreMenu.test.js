import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { MoreMenu } from '../../components/MoreMenu'

const props = {
    logout: jest.fn(),
    history: {
        push: jest.fn()
    }
}

describe('renders snapshots', () => {
    test('renders open <MoreMenu/>', () => {
        const component = renderer.create(
            <MemoryRouter>
                <MoreMenu />
            </MemoryRouter>
        );
        component.root.children[0].children[0].instance.setState({ open: true })

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders closed <MoreMenu/>', () => {
        const component = renderer.create(<MoreMenu />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('correct functions fire on button clicks', () => {
    test('closeModal fires on Import CSV button click', () => {
        const component = mount(
            <MemoryRouter>
                <MoreMenu />
            </MemoryRouter>
        );

        const menu = component.find('MoreMenu').at(0)
        menu.setState({open: true})

        expect(menu.state()).toEqual({open:true})
        component.find('[href="/dashboard/import"]').at(0).simulate('click')
        expect(menu.state()).toEqual({open:false})
    });

    test('closeModal fires on Settings button click', () => {
        const component = mount(
            <MemoryRouter>
                <MoreMenu />
            </MemoryRouter>
        );

        const menu = component.find('MoreMenu').at(0)
        menu.setState({open: true})

        expect(menu.state()).toEqual({open:true})
        component.find('[href="/dashboard/settings"]').at(0).simulate('click')
        expect(menu.state()).toEqual({open:false})
    });

    test('logout function fires on Logout button click', () => {
        const component = mount(
            <MemoryRouter>
                <MoreMenu {...props} />
            </MemoryRouter>
        );

        const menu = component.find('MoreMenu').at(0)
        menu.setState({open: true})

        component.find('button.more').at(1).simulate('click');
        expect(props.logout).toHaveBeenCalled();
        expect(props.history.push).toHaveBeenCalled();
    });
});
