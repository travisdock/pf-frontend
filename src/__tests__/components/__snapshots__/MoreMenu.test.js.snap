// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshots renders closed <MoreMenu/> 1`] = `
<div>
  <button
    className="more"
    onClick={[Function]}
  >
     More 
  </button>
</div>
`;

exports[`renders snapshots renders open <MoreMenu/> 1`] = `
<div>
  <button
    className="more"
    onClick={[Function]}
  >
     More 
  </button>
  <div
    style={
      Object {
        "left": "0px",
        "position": "absolute",
        "top": "0px",
      }
    }
  />
  <div
    className="popup-overlay "
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0,0.5)",
        "bottom": "0",
        "display": "flex",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "zIndex": "999",
      }
    }
  >
    <div
      className="popup-content "
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(255, 255, 255)",
          "border": "1px solid rgb(187, 187, 187)",
          "margin": "auto",
          "padding": "5px",
          "position": "relative",
          "width": "50%",
        }
      }
    >
      <div
        className="navbar-popup"
      >
        <a
          aria-current={null}
          href="/dashboard/import"
          onClick={[Function]}
        >
          Import CSV
        </a>
        <a
          aria-current={null}
          href="/dashboard/settings"
          onClick={[Function]}
        >
          Settings
        </a>
        <button
          className="more"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
